src/csharp/Microsoft.Spark/Sql/Types/Timestamp.cs

```diff
@@ -0,0 +1,143 @@
+        /// <param name="dateTime">DateTime object</param>
+        public Timestamp(DateTime dateTime)
+        {
+            _dateTime = dateTime.ToUniversalTime();
+
+            if (dateTime.Kind != DateTimeKind.Utc)
+            {
+                throw new InvalidTimeZoneException(
+                    $"Invalid TimeZone for Timestamp, please use " +
+                    $"Coordinated Universal Time (UTC).");
+            }
+        }
+
+        /// <summary>
+        /// Constructor for Timestamp class, the defaut timezone is
+        /// Coordinated Universal Time (UTC).
+        /// </summary>
+        /// <param name="year">The year (1 through 9999)</param>
+        /// <param name="month">The month (1 through 12)</param>
+        /// <param name="day">The day (1 through the number of days in month)</param>
+        /// <param name="hour">The hour (0 through 23)</param>
+        /// <param name="minute">The minute (0 through 59)</param>
+        /// <param name="second">The second (0 through 59)</param>
+        /// <param name="microsecond">The microsecond (0 through 999999)</param>
+        public Timestamp(int year, int month, int day, int hour, int minute, int second, int microsecond)
+        {
+            if (microsecond <= 999999 && microsecond >= 0)
+            {
+                _dateTime = new DateTime(year, month, day, hour, minute, second)
+                    .AddTicks(microsecond * 10).ToUniversalTime();
+            }
+            else
+            {
+                throw new ArgumentOutOfRangeException($"Invalid microsecond value {microsecond}");
+            }
+        }
+
+        /// <summary>
+        /// Returns the year component of the timestamp.
+        /// </summary>
+        public int Year => _dateTime.Year;
+
+        /// <summary>
+        /// Returns the month component of the timestamp.
+        /// </summary>
+        public int Month => _dateTime.Month;
+
+        /// <summary>
+        /// Returns the day component of the timestamp.
+        /// </summary>
+        public int Day => _dateTime.Day;
+
+        /// <summary>
+        /// Returns the hour component of the timestamp.
+        /// </summary>
+        public int Hour => _dateTime.Hour;
+
+        /// <summary>
+        /// Returns the minute component of the timestamp.
+        /// </summary>
+        public int Minute => _dateTime.Minute;
+
+        /// <summary>
+        /// Returns the second component of the timestamp.
+        /// </summary>
+        public int Second => _dateTime.Second;
+
+        /// <summary>
+        /// Returns the microsecond component of the timestamp.
+        /// </summary>
+        public int Microsecond => (int)(_dateTime.Ticks -
+            new DateTime(Year, Month, Day, Hour, Minute, Second).Ticks) / 10;
+
+        /// <summary>
+        /// Readable string representation for this type.
+        /// </summary>
+        public override string ToString() => _dateTime.ToUniversalTime()
```

Comment:
We should assume `_dateTime` should be already in UTC so that we don't need to call `ToUniversalTime`.

Reply:
This is the same as what happened above regarding passing kind is not working.